src/EventSubscriber/ConsoleSubscriber.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Synolia\SyliusSchedulerCommandPlugin\EventSubscriber;
+
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleSignalEvent;
+use Symfony\Component\Console\Event\ConsoleTerminateEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Synolia\SyliusSchedulerCommandPlugin\Entity\ScheduledCommand;
+use Synolia\SyliusSchedulerCommandPlugin\Enum\ScheduledCommandStateEnum;
+use Synolia\SyliusSchedulerCommandPlugin\Repository\ScheduledCommandRepository;
+
+final class ConsoleSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly ScheduledCommandRepository $scheduledCommandRepository,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ConsoleEvents::TERMINATE => 'onConsoleTerminate',
+            ConsoleEvents::SIGNAL => 'onConsoleSignal',
+        ];
+    }
+
+    public function onConsoleTerminate(ConsoleTerminateEvent $event): void
+    {
+        $this->updateCommand($event);
+    }
+
+    public function onConsoleSignal(ConsoleSignalEvent $event): void
+    {
+        $this->updateCommand($event);
+    }
+
+    private function updateCommand(ConsoleSignalEvent|ConsoleTerminateEvent $event): void
+    {
+        try {
+            $commandCode = $event->getCommand()?->getName() ?? 'no_command';
+            /** @var ScheduledCommand|null $schedulerCommand */
+            $schedulerCommand = $this->scheduledCommandRepository->findOneBy(['command' => $commandCode], ['id' => 'DESC']);
+        } catch (\Throwable) {
```

Comment:
weird that phpstan didn't yell about this :thinking: 
https://bestpractices.thecodingmachine.com/php/error_handling.html#fail-early-fail-loud

Reply:
I've restricted to Doctrine\DBAL\Exception ;)